fixtures/coverall/tests/bindings/test_coverall.rb

```diff
@@ -109,4 +113,93 @@ def test_self_by_arc
     # One reference is held by the handlemap, and one by the `Arc<Self>` method receiver.
     assert_equal coveralls.strong_count, 2
   end
+
+  def test_arcs
+    GC.start
+    coveralls = Coverall::Coveralls.new 'test_arcs'
+    assert_equal 1, Coverall.get_num_alive
+
+    assert_equal 2, coveralls.strong_count
+    assert_equal nil, coveralls.get_other
+
+    coveralls.take_other coveralls
+    # should now be a new strong ref.
+    assert_equal 3, coveralls.strong_count
+    # but the same number of instances.
+    assert_equal 1,  Coverall.get_num_alive
+    # and check it's the correct object.
+    assert_equal "test_arcs",  coveralls.get_other.get_name
+    # TODO: when we implement an identity map, it will even be the same Ruby object
+    # self.assertTrue(coveralls.get_other() is coveralls)
+
+    # TODO: using `assert_raise` seems to keep a reference to `coveralls` alive,
+    # breaking the reference-count checks further down.
```

Comment:
@saks if you have a chance to investigate this, it'd be appreciated - if I try to write the test cases here with `assert_raise` then it seems to keep the value from `coveralls` alive, interfering with the attempt to test whether its references are being cleaned up. I suspect maybe `assert_raises` is stashing the raised exception somewhere, which has a reference to the local variables including `coveralls`?

Reply:
@rfk I've experimented with it and my best guess about what happens is:
1. we define local variables (such as `coveralls`)
2. when calling `assert_raise` we pass it a block that should raise an exception
3. that block creates a closure that captures all visible local variables
4. closure holds local vars and I found no way to GC it as an object

I was able to confirm this by creating a closure before and after defining `coveralls` variable. And AFAIK this has nothing to do with `assert_raise` implementation. I was able to find lots of complaints regarding closures preventing GC from doing it's job.

Reply:
Thank you! I've updated the comment in the code to explain this a little more clearly.